### ISSUE TITLE
[charts] `onItemClick` does not work for LinePlot when used along with AreaPlot

### ISSUE THREAD

**Issue by sagar110599** [mui/mui-x#14059]
### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: (required)

### Current behavior

I have Multiple LinePlots where each of them has its area attribute to be true. OnItemClick only works for the topmost LinePlot and fails to trigger for ALL other Lines.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: OnItemClick

**Comment by michelengelen** [comment]
@JCQuintas sounds like it could be an issue with one Area overlaying all others. 👀

**Comment by michelengelen** [comment]
To help us diagnose the issue efficiently, could you provide a **stripped-down** reproduction test case using the latest version? A live example would be fantastic! ✨ 

For your convenience, our documentation offers templates and guides on creating targeted examples: [Support - Bug reproduction](https://mui.com/x/introduction/support/#bug-reproductions)

Just a friendly reminder: clean, functional code with minimal dependencies is most helpful. Complex code can make it tougher to pinpoint the exact issue.  Sometimes, simply going through the process of creating a minimal reproduction can even clarify the problem itself!

Thanks for your understanding! 🙇🏼

**Comment by sagar110599** [comment]
[screen-capture (1).webm](https://github.com/user-attachments/assets/e9cb6c10-749f-48a9-b01a-29215c6668cb)

**Comment by sagar110599** [comment]
Yes The following example involves Lines and AreaPlots both and there is a overlap. I think even though its overlap still there should be means to get the onItemClick for LinePlot right

**Comment by michelengelen** [comment]
Thanks for the video, but for us to look into the issue we would need a code reproduction.
Could you provide one please?

**Comment by sagar110599** [comment]
<ResponsiveChartContainer
        xAxis={[
          props.config.xAxis
        ]}
        yAxis={[
          props.config.yAxis
        ]}
        series={props.config?.series}
        width={props.config.width}
        height={props.config.height}
      >
        
        <BarPlot onItemClick={(event: any, identifier: any) => onChartItemClick({ 'event': event, 'identifier': identifier, 'elementType': 'bar' }, props)} />
        <LinePlot onItemClick={(event: any, identifier: any) => onChartItemClick({ 'event': event, 'identifier': identifier, 'elementType': 'line' }, props)} />
        <ChartsXAxis disableTicks={true} disableLine={true} />
        <ChartsYAxis disableTicks={true} disableLine={true} />
        <MarkPlot onItemClick={(event: any, identifier: any) => onChartItemClick({ 'event': event, 'identifier': identifier, 'elementType': 'mark' }, props)} />
        <LineHighlightPlot />
        <AreaPlot />
        <ChartsGrid horizontal={true} />
      </ResponsiveChartContainer>


Input props is as folllows
l1 = [0, 10, 14, 18, 22];
  l2 = [0, 11, 13, 16, 20];
  l3 = [0, 9, 12, 13, 15];
  config = {
    xAxis: {
      id: 'years',
      scaleType: 'point',
      data: [0, 1, 2, 3, 4, 5],
      valueFormatter: (value: any) => value + ' yr',
    },
    yAxis: {
      id: 'price',
      scaleType: 'linear',
      valueFormatter: (value: any) => value + 'k',
    },
    series: [
      {
        id: 'l1',
        data: this.l1,
        area: true,
        color: '#5B3C9C',
        type: 'line',
        label: 'L1',
        curve: 'natural',
        showMark: (dataPoint: any) => dataPoint.index === this.l1.length - 1,
      },
      {
        id: 'l2',
        data: this.l2,
        area: true,
        color: '#AA80E7',
        type: 'line',
        label: 'L2',
        curve: 'natural',
        showMark: (dataPoint: any) => dataPoint.index === this.l2.length - 1,
      },
      {
        id: 'l3',
        data: this.l3,
        area: true,
        color: '#9d7aff',
        type: 'line',
        label: 'Rent',
        curve: 'natural',
        showMark: (dataPoint: any) => dataPoint.index === this.l3.length - 1,
      },
    ],
    width: 380,
    height: 380,
  };

**Comment by sagar110599** [comment]
This is the maximum code i could post

**Comment by alexfauquette** [comment]
Maybe adding some information in the line docs saying that `<LinePlot />` needs to be on top of the `<AreaPlot />` could improve the DX

**Comment by sagar110599** [comment]
Thank you all. Sorry for the late response but the above solution works like a charm.

**Comment by michelengelen** [comment]
Great to hear @sagar110599 ... We did add this to the board already to provide some additional information in the docs regarding this! 👍🏼

**Comment by prakhargupta1** [comment]
Looks like, we need to add a demo for this and then we can close it? cc: @alexfauquette

**Comment by JCQuintas** [comment]
This is a layering effect rather than a bug. Areas are drawn as filled shapes, so an area rendered above a line intercepts the pointer before it reaches the line underneath.

The interaction layer has changed since this was reported.
